### PR TITLE
Adjustments for fresh debian/ubuntu installs

### PR DESCRIPTION
**Pull request by KickerTom**
Thank you for this module.
When using this role on a freshly created Debian Bullseye lxc container, ansible prints error due to missing networking service and then warnings about missing /etc/network and /etc/systemd/networkd.conf.d folders.
This should fix both issues.
Also, systemd in Ubuntu (tested bionic and focal) has an issue with not honoring priority of config files from /etc over /run. Network config is then hardcoded to dhcp due to /run/systemd/network/10-netplan-enp1s0.network file taking precedence.
See https://askubuntu.com/questions/1391256/systemd-networking-basics-cannot-switch-from-dhcp-to-static-ip-address

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -102,8 +120,8 @@
     follow: False
 
 - name: Check if resolvconf service exists
-  command: systemctl list-unit-files resolvconf.service
-  failed_when: False
+  command: systemctl cat resolvconf.service
+  failed_when: resolvconf_service_list.rc not in [0, 1]
```
Comment (haslersn) [inline tasks/main.yml:111]: Why this change?

**Review by haslersn — CHANGES_REQUESTED** [review]
I think the changes regarding the names of network files should be in a different PR. I'm not sure what and why you're doing there.

File: tasks/main.yml (hunk 1 above)
Comment (KickerTom) [reply tasks/main.yml:111]: It's been a while, so I may not remember the reason correctly, but I believe the former approach didn't set the return code of the command on some systems, so the following test on line 134 will not work.
Just tested it:
on debian 11.2:
```
$ systemctl list-unit-files nonexistent.service > /dev/null 2>&1 ; echo $?
1
$ systemctl cat nonexistent.service > /dev/null 2>&1 ; echo $?
1
```
on debian 10.13:
```
$ systemctl list-unit-files nonexistent.service > /dev/null 2>&1 ; echo $?
0
$ systemctl cat nonexistent.service > /dev/null 2>&1 ; echo $?
1
```

**Comment by KickerTom** [comment]
Sure, I can split the PR. Only the two last commits should be split, or something else as well?

**Comment by haslersn** [comment]
> Sure, I can split the PR. Only the two last commits should be split, or something else as well?

Yes, I think if the last two commits are removed we can merge this PR.

File: tasks/main.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
 ---
+- name: Create extra directories relevant for systemd-networkd
+  file:
+    path: "{{ item | dirname }}"
+    state: directory
+  loop: "{{ systemd_network_copy_files | map(attribute='dest') | list + ['/etc/systemd/networkd.conf.d/', '/etc/systemd/resolved.conf.d/', '/etc/network/'] }}"
```
Comment (haslersn) [inline tasks/main.yml:6]: Why `/etc/network`?

Comment (KickerTom) [reply tasks/main.yml:6]: Because at the end of the file, you add a dummy /etc/network/interfaces file there explaining the network is now systemd managed. That part fails (stops ansible play) if the directory does not exists. Maybe would be better create that dummy file only if the folder exists? Not sure...

**Review by haslersn — APPROVED** [review]
